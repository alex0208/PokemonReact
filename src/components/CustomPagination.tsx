import { Pagination } from '@mui/material';
import { ChangeEvent, Dispatch, SetStateAction } from 'react';

interface Props {
    currentPage: number;
    setCurrentPage: Dispatch<SetStateAction<number>>;
}

const CustomPagination = ({ currentPage, setCurrentPage }: Props) => {
    const handlePageChange = (e: ChangeEvent<unknown>, value: number) => {
        setCurrentPage(value);
        window.scrollTo(0, 0);
    };
    return (
        <Pagination
            sx={{ my: 3 }}
            count={currentPage + 2}
            siblingCount={3}
            boundaryCount={2}
            page={currentPage}
            onChange={handlePageChange}
        />
    );
};

export default CustomPagination;
