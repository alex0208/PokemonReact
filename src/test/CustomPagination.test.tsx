import { Dispatch, SetStateAction } from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import CustomPagination from '../components/CustomPagination';

it('matches snapshot', () => {
    const setter: Dispatch<SetStateAction<number>> = () => {};

    const tree = renderer
        .create(
            <MemoryRouter>
                <CustomPagination currentPage={1} setCurrentPage={setter} />
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
