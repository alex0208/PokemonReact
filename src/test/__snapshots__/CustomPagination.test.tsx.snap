// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  aria-label="pagination navigation"
  className="MuiPagination-root-jiVNev dslAHc MuiPagination-root MuiPagination-text"
>
  <ul
    className="MuiPagination-ul-AaxlE VIygi MuiPagination-ul"
  >
    <li>
      <button
        aria-label="Go to previous page"
        className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root Mui-disabled MuiPaginationItem-root-eKOth cwlSWf MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root-ffGNMi ufROg MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon-cIGteT eNVKWa MuiPaginationItem-icon"
          data-testid="NavigateBeforeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </button>
    </li>
    <li>
      <button
        aria-current="true"
        aria-label="page 1"
        className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root MuiPaginationItem-root-eKOth cwlSWf MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        1
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 2"
        className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root MuiPaginationItem-root-eKOth cwlSWf MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        2
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 3"
        className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root MuiPaginationItem-root-eKOth cwlSWf MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        3
      </button>
    </li>
    <li>
      <button
        aria-label="Go to next page"
        className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root MuiPaginationItem-root-eKOth cwlSWf MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root-ffGNMi ufROg MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon-cIGteT eNVKWa MuiPaginationItem-icon"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </button>
    </li>
  </ul>
</nav>
`;
